src/Executable/Executable.php

```diff
@@ -0,0 +1,44 @@
+<?php
+
+namespace Castor\Executable;
+
+use Castor\Container;
+use Castor\Context;
+use Symfony\Component\Process\Process;
+
+use function Castor\context;
+
+abstract class Executable implements ExecutableInterface
+{
+    /** @var (callable(Context): Context)[] */
+    protected array $contextModifiers = [];
+
+    /**
+     * @param (callable(string, string, Process): void)|null $callback
+     */
+    public function run(?Context $context = null, ?callable $callback = null): Process
+    {
+        $context ??= context();
+
+        foreach ($this->contextModifiers as $modifier) {
+            $context = $modifier($context);
+        }
+
+        return Container::get()->processRunner->run(
+            $this->getCommand(),
+            callback: $callback,
+            context: $context,
+        );
+    }
+
+    /**
+     * @param (callable(Context): Context) $modifier
+     */
+    protected function addContextModifier(callable $modifier): void
```

Comment:
This is not used anywhere. So is it really useful? If yes, could you add an example with it (like that it'll be tested)

Reply:
In fact it should be internal for now